FormSubmissionAndValidation2/src/components/DropdownBox.jsx

```diff
@@ -0,0 +1,36 @@
+import PropTypes from "prop-types";
+import { ErrorMessage } from "./ErrorMessage";
+import { userFormConfig } from "../helper/userHelper";
+import React from "react";
+
+export const DropDownBox = React.memo(function DropDownBox({ fieldId, options, value, errorMessage, updateValue }) {
+  const label = userFormConfig[fieldId].label;
```

Comment:
Similar issue with proptypes not lining up with expected behavior. What happens if fieldId is nil or if the specific entry does not exist?

Reply:
made the corresponding fields required.